std/zip.d

```diff
@@ -726,6 +741,10 @@ public:
             auto localFileHeaderNamelen = getUshort(de.offset + 26);
             auto localFileHeaderExtralen = getUshort(de.offset + 28);
 
+            // file data
+            removeSegment(de.offset, de.offset + localFileHeaderLength + localFileHeaderNamelen
+                                     + localFileHeaderExtralen + de._compressedSize);
```

Comment:
I'm hoping all of these can be later refactored away to avoid the DRY of calculating all these lengths all over the place. The code is pretty difficult to follow as it is. Maybe once the structures are represented as D structs, the `removeSegment` call can be done by whatever function will be doing the reading of the entire structs.

Reply:
Here is my code to read a Windows PE file's headers:

https://github.com/CyberShadow/ae/blob/218c29e17f6ecfc49f696b92c879c5c6b9917f2e/sys/windows/pe/pe.d#L31-L51

I'm hoping std.zip could closer to that.